api-1.11/src/main/java/org/openmrs/module/fhir/helper/AllergyHelperImpl1_11.java

```diff
@@ -0,0 +1,104 @@
+import org.openmrs.module.fhir.api.util.FHIRUtils;
+import org.openmrs.module.fhir.util.FHIRAllergyIntoleranceUtil1_11;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Component(value = "fhir.AllergyHelper")
+@OpenmrsProfile(openmrsVersion = "1.11.*")
+public class AllergyHelperImpl1_11 implements AllergyHelper {
+
+	@Autowired
+	private PatientService patientService;
+
+
+	@Override
+	public AllergyIntolerance getAllergyIntolerance(String uuid) {
+		Allergy allergy = (Allergy) Context.getActiveListService().getActiveListItemByUuid(uuid);
+		return allergy != null ?
+				FHIRAllergyIntoleranceUtil1_11.generateAllergyIntolerance(allergy)
+				: null;
+	}
+
+	@Override
+	public Collection<AllergyIntolerance> getAllergyIntoleranceByPatient(Patient patient) {
+		List<AllergyIntolerance> allergies = new ArrayList<>();
+		PatientService allergyService = Context.getService(PatientService.class);
+		for (Allergy allergy : allergyService.getAllergies(patient)) {
+			allergies.add(FHIRAllergyIntoleranceUtil1_11.generateAllergyIntolerance(allergy));
+		}
+		return allergies;
+	}
+
+	@Override
+	public AllergyIntolerance createAllergy(AllergyIntolerance allergyIntolerance) {
+		List<String> errors = new ArrayList<String>();
+		Allergy allergy = FHIRAllergyIntoleranceUtil1_11.generateAllergy(allergyIntolerance, errors);
+		FHIRUtils.checkGeneratorErrorList(errors);
+		return FHIRAllergyIntoleranceUtil1_11.generateAllergyIntolerance(saveAllergy(allergy));
+	}
+
+	@Override
+	public AllergyIntolerance updateAllergy(AllergyIntolerance allergyIntolerance, String uuid) {
+		List<String> errors = new ArrayList<String>();
+		Allergy newAllergy = FHIRAllergyIntoleranceUtil1_11.generateAllergy(allergyIntolerance, errors);
+		FHIRUtils.checkGeneratorErrorList(errors);
+
+		Allergy allergy = (Allergy) Context.getActiveListService().getActiveListItemByUuid(uuid);
+		if (allergy != null) {
+			allergy = FHIRAllergyIntoleranceUtil1_11.updateAllergyAttributes(allergy, newAllergy);
+			allergy = saveAllergy(allergy);
+		} else {
+			newAllergy.setUuid(uuid);
+			allergy = saveAllergy(newAllergy);
+		}
+
+		return FHIRAllergyIntoleranceUtil1_11.generateAllergyIntolerance(allergy);
+	}
+
+	@Override
+	public void deleteAllergy(String uuid) {
+		Allergy allergy = (Allergy) Context.getActiveListService().getActiveListItemByUuid(uuid);
+
+		if (allergy == null) {
+			throw new ResourceNotFoundException(String.format("Allergy with id '%s' not found", uuid));
+		} else {
+			try {
+				patientService.removeAllergy(allergy, FHIRConstants.FHIR_RETIRED_MESSAGE);
+			} catch (APIException apie) {
+				throw new MethodNotAllowedException(String.format("OpenMRS has failed to retire allergy'%s': %s", uuid,
+						apie.getMessage()));
+			}
+		}
+	}
+
+	@Override
+	public Object generateAllergy(Object object) {
```

Comment:
can we use here Allegry instead of Object

Reply:
We can't because between versions of the OpenMRS platform the Allergy class package was changed.